File: ionic/src/interfaces/registration.ts
import {Country, Text, Service, PrivacyOption, NameStructure, Address} from "./app";

//Responses
export interface ResGetReady {
  successfully : boolean,
  data : {
    countries : Country [],
    texts : Text,
    services : Service [],
    privacyOptions: PrivacyOption []
  }
}
export interface ResSendSMS {
  successfully : boolean,
  data : null
}
export interface ResCheckAvailability {
  successfully : boolean,
  data : {
    isAvailable: boolean
  }
}
export interface ResConfirmSMS {
  successfully : boolean,
  data : {
    uuid: string
  }
}
export interface ResRegister {
  successfully : boolean,
  data : null
}


//Requests
export interface ReqConfirmSMS {
  code : string,
  mobileNumber : string
}
export interface ReqRegister {
  username : string,
  phoneUUID : string,
  phoneCode : string,
  picture : string,
  password : string,
  name : NameStructure,
  birthDay: Date,
  bio : string,
  gender : string,
  address : Address,
  services: string [],
  email : string,
  privacyOptions : PrivacyOption [],
  tos : string,
  customData : any
}

File: xamarin_forms/xamarin_forms/Models/Registration.cs
using System;
using System.Collections.Generic;
namespace xamarin_forms
{
    public class Data_getReady
    {
        public List<Country> countries { get; set; }
        public Texts texts { get; set; }
        public List<Venue> venues { get; set; }
        public List<PrivacyOptions> privacyOptions { get; set; }
    }
    public class ResGetReady
    {
        public bool successfully { get; set; }
        public Data_getReady data { get; set; }
    }
}
